### PR TITLE
Refactor markdown renderer

### PR DESCRIPTION
**Pull request by Aayyush**
The current markdown renderer is overloaded with a lot of conditional statements and redundant methods making it hard to follow. So in this PR, I have refactored out the logic for resolving templates into a TemplateResolver that also deals with template overrides. 

Any suggestions to improve it further are welcome :)

### CODE REVIEW

**Review by msarvar — COMMENTED** [review]
I like the changes, one suggestion. If we gonna refactor rendering part we should move it out of events package.

**Comment by Aayyush** [comment]
> I like the changes, one suggestion. If we gonna refactor rendering part we should move it out of events package.

Hmm I agree! I was thinking of moving the `markdown_renderer.go` and `template_resolver.go` into the templates package inside the events directory. If we want to move it out of events package entirely, we could put the templates package in the server directory. What do you think?

File: server/events/template_resolver.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+		tmpl = t.getApplyTmpl(templateOverrides, numPrjResults)
+	case common.Command == versionCommandTitle:
+		tmpl = t.getVersionTmpl(templateOverrides, common, numPrjResults, numVersionSuccesses)
+	}
+
+	return tmpl
+}
+
+// Resolves templates for project commands
+func (t *TemplateResolver) ResolveProject(result command.ProjectResult, baseRepo models.Repo, common commonData) (*template.Template, interface{}) {
```
Comment (nishkrishnan) [inline server/events/template_resolver.go:103]: so is this the function we'd call for the checks case?

**Review by nishkrishnan — COMMENTED** [review]
Nice job, i agree that we should move this out of events since we're refactoring it anyways.  What if we put this in `server/vcs/markdown`? Is that possible? This is really only needed for vcs anyways, our webui logs are rendered completely different anyways.

File: server/events/template_resolver.go (hunk 1 above)
Comment (Aayyush) [reply server/events/template_resolver.go:103]: Yup to render each project in their own checkrun output

**Comment by Aayyush** [comment]
> Nice job, i agree that we should move this out of events since we're refactoring it anyways. What if we put this in `server/vcs/markdown`? Is that possible? This is really only needed for vcs anyways, our webui logs are rendered completely different anyways.

Yup, I think that makes sense! I have moved the templating stuff to `server/vcs/markdown`

**Review by msarvar — APPROVED** [review]
good job cleaning up
